framework/src/utils/MooseMeshUtils.C

```diff
@@ -571,4 +571,45 @@ extraElemIntegerSwapParametersProcessor(
                                     elem_integer_swap_pairs.end());
   }
 }
+
+std::unique_ptr<ReplicatedMesh>
+buildBoundaryMesh(const MeshBase & input_mesh, const boundary_id_type boundary_id)
+{
+  auto poly_mesh = std::make_unique<ReplicatedMesh>(input_mesh.comm());
```

Comment:
we should probably take in a ReplicatedMesh reference instead

Reply:
Why? I think this is a cleaner interface.

Reply:
does this operation serialize the mesh?

Reply:
No, because the input mesh is a replicated mesh. I actually changed the input mesh type from `MeshBase` to `ReplicatedMesh` to avoid confusion.